project/src/CMakeListsTxt.cpp

```diff
@@ -0,0 +1,227 @@
+#include "../../utils/include/utils.h"
+#include "Configuration.h"
+#include "Project.h"
+
+#include <boost/filesystem/fstream.hpp>
+#include <ostream>
+
+struct CMakeProjectExporter
+{
+    explicit CMakeProjectExporter(const Project &project) :
+        project_{project}
+    {
+    }
+
+    std::string LookupLibraryName(const std::string &componentName)
+    {
+        return (project_.IsSystemComponent(componentName) ? cmakeSystemProjectPrefix + componentName : componentName);
+    }
+
+    void createCMakeListsFiles(const GlobalOptions &opts)
+    {
+        auto injectOptions = [](std::ostream &os, const std::string &target_option, const std::string &target, const std::string &access, const std::vector<std::string> &opts) {
+            if(!opts.empty())
+            {
+                os << target_option << "(" << target << " " << access;
+                for(const auto &opt : opts)
+                {
+                    os << " " << opt;
+                }
+                os << ")\n";
+            }
+        };
+
+        for(const auto &comp : project_.components)
+        {
+            boost::filesystem::path filePath = project_.projectRoot / comp.second.root / cmakelists_txt;
+            boost::system::error_code ec;
+            boost::filesystem::remove(filePath, ec);
+            boost::filesystem::ofstream os{filePath};
+
+            std::string target = comp.second.GetName();
+
+            dumpDoNotModifyWarning(os);
+
+            if(comp.second.type != "library" && comp.second.type != "executable")
+            {
+                continue;
+            }
+            std::string target_type = "add_executable(";
+            std::string target_modifier;
+            std::string publicAccess = "PUBLIC";
+            std::string privateAccess = "PRIVATE";
+
+            if(comp.second.type == "library")
+            {
+                target_type = "add_library(";
+                target_modifier = " STATIC";
+
+                if(comp.second.isHeaderOnly())
+                {
+                    target_modifier = " INTERFACE";
+                    publicAccess = "INTERFACE";
+                    privateAccess = "INTERFACE"; // We probably should just abort here
```

Comment:
Can you change the comment to indicate why this is?

Reply:
I'll clarify the comment.  That was in regards to a header only library having a private dependency.  It just doesn't make sense.